.travis.yml

```diff
@@ -8,51 +8,104 @@ addons:
       - mysql-client-core-5.6
       - mysql-client-5.6
       - postfix
-  firefox: latest
+  chrome: stable
   hosts:
     - magento2.travis
 services:
   - rabbitmq
   - elasticsearch
+  - mysql
 language: php
 php:
   - '7.1'
   - '7.2'
   - '7.3'
 env:
   global:
-    - COMPOSER_BIN_DIR=~/bin
+    # we should revisit the nodejs version. Long Term Stable (LTS) is version 12 now.
     - NODE_JS_VERSION=8
     - MAGENTO_HOST_NAME="magento2.travis"
+    - MAGENTO_PROTOCOL="https"
+    - MAGENTO_BACKEND="backend"
+    - MAGENTO_ADMIN_USERNAME="admin"
+    - MAGENTO_ADMIN_PASSWORD="123123q"
   matrix:
     - TEST_SUITE=unit
     - TEST_SUITE=phpstan
     - TEST_SUITE=static
+    # List out subsets of functional tests to run separately to fit under travis timeout (50mins)
+    # TODO: this is hacky, is there a better way?
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_filters_part_one
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_filters_part_two
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_grid
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_preview
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_configuration
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_keywords
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_ims
+    - TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_see_more
 matrix:
   fast_finish: true
+jobs:
+  # only run MFTF tests on PHP 7.3
+  exclude:
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_filters_part_one
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_filters_part_two
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_grid
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_preview
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_configuration
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_keywords
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_ims
+    - php: '7.1'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_see_more
+    - php: '7.2'
+      env: TEST_SUITE=functional MFTF_SUITE=adobe_stock_integration_suite_filters_part_one
```

Comment:
The fix of the issue slowing down the tests is now merged to `2.3-develop`. Could you check if we could get rid of `part_*` test suites on the latest code base?